Lib/sre_parse.py

```diff
@@ -260,6 +261,14 @@ def _class_escape(source, escape):
         elif c in DIGITS:
             raise error, "bogus escape: %s" % repr(escape)
         if len(escape) == 2:
+            if sys.py3kwarning and c in ASCIILETTERS:
+                import warnings
+                if c in 'Uu' and state.flags & SRE_FLAG_UNICODE:
+                    warnings.warn('unicode escape %s' % escape,
```

Comment:
This warning is going to be shown mostly to people who did want a Unicode escape.
See e.g.: https://github.com/sphinx-doc/sphinx/issues/2544 https://github.com/translate/translate/issues/3449  https://github.com/amperser/proselint/issues/672 https://github.com/python-babel/babel/issues/472
Unfortunately, the warning message doesn't give much clue about what's wrong in this case.